### DISCUSSION TITLE
Is it possible to inject custom code into the `__init__.py` of the wheel for a binary extension module?

### DISCUSSION THREAD

**Discussion by ghyatzo** [PyO3/maturin#2212] [Q&A] [answered]
basically the title.
I have a python module written in rust with bindings from pyo3, and I successfully build the wheels with maturin. Everything Just Works, but i was wondering If I could specify somehow some code to run upon `import`ing my module, usually that is done in the `__init__.py` and I can manually do it by extracting the wheel modifying the file and repackage it.
Is there a way to do it automatically?

**Comment by messense** [comment, marked as answer]
You can switch to a mixed Rust/Python layout to have full control over the code in `__init__.py`, see https://www.maturin.rs/project_layout#mixed-rustpython-project.
